product_price_category/models/product_pricelist.py

```diff
@@ -0,0 +1,249 @@
+                                                  round=False)
+                else:
+                    # if base option is public price take sale price
+                    # else cost price of product
+                    # price_get returns the price in the context UoM,
+                    #  i.e. qty_uom_id
+                    price = product.price_compute(rule.base)[product.id]
+
+                convert_to_price_uom = (
+                    lambda price: product.uom_id._compute_price(
+                        price, self.env['product.uom'].browse(price_uom_id)
+                    )
+                )
+
+                if price is not False:
+                    if rule.compute_price == 'fixed':
+                        price = convert_to_price_uom(rule.fixed_price)
+                    elif rule.compute_price == 'percentage':
+                        price = (price -
+                                 (price * (rule.percent_price / 100))) or 0.0
+                    else:
+                        # complete formula
+                        price_limit = price
+                        price = (price -
+                                 (price * (rule.price_discount / 100))) or 0.0
+                        if rule.price_round:
+                            price = tools.float_round(
+                                price, precision_rounding=rule.price_round
+                            )
+
+                        if rule.price_surcharge:
+                            price_surcharge = convert_to_price_uom(
+                                rule.price_surcharge
+                            )
+                            price += price_surcharge
+
+                        if rule.price_min_margin:
+                            price_min_margin = convert_to_price_uom(
+                                rule.price_min_margin
+                            )
+                            price = max(price, price_limit + price_min_margin)
+
+                        if rule.price_max_margin:
+                            price_max_margin = convert_to_price_uom(
+                                rule.price_max_margin
+                            )
+                            price = min(price, price_limit + price_max_margin)
+                    suitable_rule = rule
+                break
+            # Final price conversion into pricelist currency
+            if suitable_rule and suitable_rule.compute_price != 'fixed' \
+                    and suitable_rule.base != 'pricelist':
+                price = product.currency_id.compute(
+                    price, self.currency_id, round=False
+                )
+
+            results[product.id] = (
+                price, suitable_rule and suitable_rule.id or False
+            )
+        return results
+
+
+class ProductPricelistItem(models.Model):
+    _inherit = 'product.pricelist.item'
+
+    applied_on = fields.Selection(
+        selection_add=[('2b_product_price_category', 'Price Category')]
```

Comment:
Hi, 
why there is the prefix "2b_" ?

Reply:
@legalsylvain 

as it is an alternative for the product_category it makes sense to have `2b` instead of 4

``` python
    applied_on = fields.Selection([
        ('3_global', 'Global'),
        ('2_product_category', ' Product Category'),
        ('1_product', 'Product'),
        ('0_product_variant', 'Product Variant')], "Apply On",
        default='3_global', required=True,
        help='Pricelist Item applicable on selected option')

```